### ISSUE TITLE
Updating a Cosmos DB account with duplicate read & write replication regions takes 10 minutes

### ISSUE THREAD

**Issue by alulyazan** [Azure/azure-libraries-for-net#164]
If you call Define & Create on a Cosmos DB account with the same parameters (which should return relatively quickly since nothing is being created), and the Write and Read replication regions are the same, it takes 10 minutes to complete. Looking in fiddler, it checks the status of the operation every 5 seconds, even though the status on each of the read and write replication regions and the failover are "succeeded". 

In the source code, it looks like it will loop if the failover policies count is less than the count of the regions specified, which is not deduped in the SDK.

Sample code:

```
Region region = Region.Create("South Central US");
await _azure.CosmosDBAccounts.Define(cdbName)
                .WithRegion(region)
                .WithExistingResourceGroup(rg)
                .WithKind(DatabaseAccountKind.GlobalDocumentDB)
                .WithSessionConsistency()
                .WithWriteReplication(region)
                .WithReadReplication(region)
                .CreateAsync(cancellationToken);
```

**Comment by hovsepm** [comment]
@alulyazan I assume you are referring to the source code here - https://github.com/Azure/azure-libraries-for-net/blob/master/src/ResourceManagement/CosmosDB/CosmosDBAccountImpl.cs#L59

**Comment by alulyazan** [comment]
That's correct

**Comment by hovsepm** [comment]
@alulyazan tcould you also clarify what you mean by "which is not deduped in the SDK"?

**Comment by alulyazan** [comment]
Just that in my example, in the above file in line 68, createUpdateParametersInner.Locations has two entries with a location of South Central US, but the response on line 69 contains one failover policy, so the loop never exits. I was just making an assumption on a fix, which is to get a count of the unique locations in createUpdateParametersInner.Locations.

**Comment by hovsepm** [comment]
Thanks for the explanation.

**Comment by milismsft** [comment]
This is fixed in version 1.11.1.
